homeassistant/components/faadelays/binary_sensor.py

```diff
@@ -0,0 +1,104 @@
+"""Platform for FAA Delays sensor component."""
+from homeassistant.components.binary_sensor import BinarySensorEntity
+
+from .const import DOMAIN, FAA_BINARY_SENSORS
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up a FAA sensor based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors = []
+    for kind, attrs in FAA_BINARY_SENSORS.items():
+        name, icon = attrs
```

Comment:
I'm a little confused here, could use some more guidance. I'm happy to fix it but not quite sure what you mean! I currently have a dictionary which defines each type of sensor in `const.py` and I'm just iterating through it here to define each type of sensor. Do you mean the rest of the attributes for each type (lines 69-88) should also be in a dictionary rather than individually defined?

Reply:
Well, now its a set with 2 positions. name & icon. The problem is, if you start adding more things, such a fixed set is harder to extend.

So instead:

```py
FAA_BINARY_SENSORS = {
    "GROUND_DELAY": {
      ATTR_NAME: "Ground Delay",
      ATTR_ICON: "mdi:airport",
    },
    ...
}
```

And use `attr[ATTR_ICON]` here instead.

In the future, if you wanted to add `ATTR_DEVICE_CLASS` to some entities, it could simply be added to the existing dict.

Reply:
Got it, that makes sense. I do like that better as well. I'll bring that in in another commit.

Reply:
Completed!